### PR TITLE
Adds Rails matcher for AST-scanners

### PR DESCRIPTION
**Pull request by davidwessman**
- AST-matching is extracted to a class.
- Adds a class for matching Rails model translations.

### CODE REVIEW

**Comment by glebm** [comment]
Shall we merge this before #436?

**Comment by davidwessman** [comment]
> Shall we merge this before #436?

No, I think we should do the comments first. Will see if I can get some more time to work on it today.

File: lib/i18n/tasks/scanners/ruby_ast_scanner.rb [hunk 1]
```diff
@@ -131,7 +132,7 @@ def setup_matchers
             message: message,
             scanner: self
           )
-        end
+        end << AstMatchers::RailsModelMatcher.new(scanner: self)
```
Comment (davidwessman) [inline lib/i18n/tasks/scanners/ruby_ast_scanner.rb:135]: @glebm Should we check some config if the rails-matcher should be used?

File: lib/i18n/tasks/scanners/ast_matchers/rails_model_matcher.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      children = Array(send_node&.children)
+      return unless children[1] == :human
+
+      base_children = Array(children[0]&.children)
+      class_node = base_children[0]
+
+      return unless class_node&.type == :const && base_children[1] == :model_name
+
+      model_name = class_node.to_a.last.downcase
+      key = "activerecord.models.#{model_name}.other"
```
Comment (davidwessman) [inline lib/i18n/tasks/scanners/ast_matchers/rails_model_matcher.rb:53]: I tried to check how pluralization is done in other places, not really sure which key to use here or if we somehow can mark multiple different keys?

Comment (glebm) [reply lib/i18n/tasks/scanners/ast_matchers/rails_model_matcher.rb:53]: Pluralization is handled elsewhere, here it should probably be "activerecord.models.#{model_name}"

File: lib/i18n/tasks/scanners/ruby_ast_scanner.rb (hunk 1 above)
Comment (glebm) [reply lib/i18n/tasks/scanners/ruby_ast_scanner.rb:135]: Yeah, let's think of a way to make this configurable

File: lib/i18n/tasks/scanners/ruby_ast_scanner.rb [hunk 3]
```diff
@@ -132,6 +133,10 @@ def setup_matchers
             scanner: self
           )
         end
+        if Array(config[:extra_matchers]).include?('rails_model')
```
Comment (glebm) [inline lib/i18n/tasks/scanners/ruby_ast_scanner.rb:136]: Perhaps the config should be a list of class names? This way it is extensible outside of i18n-tasks

Comment (davidwessman) [reply lib/i18n/tasks/scanners/ruby_ast_scanner.rb:136]: Hm, yes that makes sense. Like the scanners works?

Comment (glebm) [reply lib/i18n/tasks/scanners/ruby_ast_scanner.rb:136]: Yeah

Comment (davidwessman) [reply lib/i18n/tasks/scanners/ruby_ast_scanner.rb]: Done 🙂

**Review by glebm — APPROVED** [review]

**Pull request merged**